### ISSUE TITLE
Add input validation and guidance to the metadata editor

### ISSUE THREAD

**Issue by zbodi74** [metabase/metabase#49763]
**Is your feature request related to a problem? Please describe.**

Currently, the metadata editor allows column settings to be configured without much guidance or validation, which can make configuration error-prone — especially for less experienced users. Since many features rely on metadata settings, misconfigurations can lead to unexpected or confusing behavior which is challenging to troubleshoot.

**Describe the solution you'd like**

Implement a set of validation rules based on common, best-practice assumptions for metadata configuration. 
When a table is opened in the editor, Metabase should use these rules to perform checks on column-level settings and highlight configurations that may lead to issues.

For instance, warn users if there are multiple columns assigned the "Entity Name" field type. We expect only a single column with this setting (to display the entity name in field filter drop-downs), and a warning would help users avoid setting multiple columns which this type, which currently leads to undefined behavior.

**Describe alternatives you've considered**
n/a

**How important is this feature to you?**
It would help both new and experienced users configure metadata more intuitively and correctly, reducing confusion and improving the overall experience.

**Comment by ranquild** [comment]
@thebiglabasky some things that come to mind:
- `Entity Name` can only be used once per table
- most semantic types are useless, not sure if it's even a validation problem

**Comment by thebiglabasky** [comment]
Suggested validations:<br><br>**Field Type Incompatibilities**

* **Entity Name**: Warn when multiple columns in a table are assigned the "Entity Name" field type (only one column should have this designation)


* **Entity Key**: Warn when multiple columns in a table are assigned the "Entity Key" field type


* **Foreign Key**: Warn when a column is set as a Foreign Key but no target field is selected


* **Foreign Key Target**: Warn when a Foreign Key points to a field that is not an Entity Key in the target table


* **Coordinate Fields**: Warn when a Latitude field exists without a corresponding Longitude field in the same table (and vice versa)


* **Currency Fields**: Warn when a field is set to a financial field type but no currency type is selected

## **Visibility Incompatibilities**

* **Critical Fields**: Warn when critical fields (Entity Key, Entity Name, or Foreign Keys) are set to "Do not include" visibility


* **(Stretch) Related Fields**: Warn when one part of a related field pair (like Latitude/Longitude) is visible but the other is hidden

## **Filter Widget Incompatibilities**

* **Dropdown Lists**: Warn when a field's filter widget is set to "A list of all values" but the field has too many distinct values (>1,000)

**Comment by ranquild** [comment]
@thebiglabasky <br><br>**Entity Key** - multiple `Entity Key` is fine, but with multiple PKs the object detail view won't work. But we should not forbid them.<br><br>**Foreign Key Target** - I think it should be impossible to configure it this way (initially). But later indeed the FK target can changed to a non-PK.<br><br>**Coordinate Fields** - overall good catch, but also this is not blocking

**Comment by thebiglabasky** [comment]
Yeah coordinates is defo 🍒 on top.<br>Adjusting the proposal on Entity Key and clarifying the FK one 👍
